app/src/main/java/net/squanchy/notification/NotificationCreator.kt

```diff
@@ -0,0 +1,203 @@
+                .setTicker(
+                        context.resources.getQuantityString(
+                                R.plurals.event_notification_ticker,
+                                talksCount,
+                                talksCount
+                        )
+                )
+                .setDefaults(Notification.DEFAULT_SOUND or Notification.DEFAULT_VIBRATE)
+                .setLights(
+                        ContextCompat.getColor(context, R.color.notification_led_color),
+                        NOTIFICATION_LED_ON_MS,
+                        NOTIFICATION_LED_OFF_MS
+                )
+                .setSmallIcon(R.drawable.ic_stat_notification)
+                .setAutoCancel(true)
+                .extend(extender)
+    }
+
+    private fun createPendingIntentForSingleEvent(eventId: String): PendingIntent {
+        val taskBuilder = createBaseTaskStackBuilder()
+        val eventDetailIntent = EventDetailsActivity.createIntent(context, eventId)
+        taskBuilder.addNextIntent(eventDetailIntent)
+
+        return taskBuilder.getPendingIntent(0, PendingIntent.FLAG_CANCEL_CURRENT)
+    }
+
+    private fun getPlaceName(event: Event): String? {
+        return if (event.place.isPresent) event.place.get().name else null
+    }
+
+    private fun getTrackColor(event: Event): Int {
+        return event.track
+                .map { (_, _, accentColor) -> Color.parseColor(accentColor.or(ARGB_TRANSPARENT)) }
+                .or(Color.TRANSPARENT)
+    }
+
+    private fun createPendingIntentForMultipleEvents(): PendingIntent {
+        val taskBuilder = createBaseTaskStackBuilder()
+        return taskBuilder.getPendingIntent(0, PendingIntent.FLAG_CANCEL_CURRENT)
+    }
+
+    private fun createBaseTaskStackBuilder(): TaskStackBuilder {
+        val homescreenIntent = Intent(context, HomeActivity::class.java)
+        homescreenIntent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
+        return TaskStackBuilder.create(context)
+                .addNextIntent(homescreenIntent)
+    }
+
+    private fun createBigTextRichNotification(notificationBuilder: NotificationCompat.Builder, event: Event): NotificationCompat.BigTextStyle {
+        val bigTextBuilder = StringBuilder()
+                .append(getSpeakerNamesFrom(event.speakers))
+
+        val placeName = getPlaceName(event)
+        if (!placeName.isNullOrEmpty()) {
+            bigTextBuilder
+                    .append('\n')
+                    .append(context.getString(R.string.event_notification_starting_in, placeName))
+        }
+
+        return NotificationCompat.BigTextStyle(notificationBuilder)
+                .setBigContentTitle(event.title)
+                .bigText(bigTextBuilder.toString())
+    }
+
+    private fun getSpeakerNamesFrom(speakers: List<Speaker>): String {
+        val speakerNames = TextUtils.join(", ", map<Speaker, String>(speakers, { it.name }))
```

Comment:
You can simplify this using [`List.joinToString()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/join-to-string.html)

Reply:
Nice 🎉